### ISSUE TITLE
iwlwifi and Bluetooth: hci0  firmware fails to load with kernel 5.2.14

### ISSUE THREAD

**Issue by mudrii** [NixOS/nixpkgs#68770]
**Describe the bug**
NixOs latest kernel can load wifi and Bluetooth modules.

**To Reproduce**
Install NIxOs 19.03 on laptop Lenovo p53 workstation with AX200 Wi-Fi 6 802.11AX 

## dmesg oputput
```sh
* [   11.212346] Intel(R) Wireless WiFi driver for Linux
  [   11.212346] Copyright(c) 2003- 2015 Intel Corporation
* [   11.212601] iwlwifi 0000:52:00.0: enabling device (0000 -> 0002)
* [   11.225446] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-48.ucode faile* d with error -2
* [   11.225453] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-47.ucode faile* d with error -2
* [   11.225458] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-46.ucode faile* d with error -2
* [   11.225462] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-45.ucode faile* d with error -2
* [   11.225466] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-44.ucode faile* d with error -2
* [   11.225469] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-43.ucode faile* d with error -2
* [   11.225473] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-42.ucode faile* d with error -2
* [   11.225477] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-41.ucode faile* d with error -2
* [   11.225481] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-40.ucode faile* d with error -2
* [   11.225485] iwlwifi 0000:52:00.0: Direct firmware load for iwlwifi-cc-a0-39.ucode faile* d with error -2
* [   11.225486] iwlwifi 0000:52:00.0: no suitable firmware found!
* [   11.226515] iwlwifi 0000:52:00.0: minimum version required: iwlwifi-cc-a0-39
* [   11.226993] iwlwifi 0000:52:00.0: maximum version supported: iwlwifi-cc-a0-48
* [   11.227647] iwlwifi 0000:52:00.0: check git://git.kernel.org/pub/scm/linux/kernel/git/f* irmware/linux-firmware.git
  [   11.228717] usbcore: registered new interface driver btusb
  [   11.228933] Bluetooth: hci0: Bootloader revision 0.3 build 0 week 24 2017
  [   11.229950] Bluetooth: hci0: Device revision is 1
  [   11.229951] Bluetooth: hci0: Secure boot is enabled
  [   11.229951] Bluetooth: hci0: OTP lock is enabled
  [   11.229951] Bluetooth: hci0: API lock is enabled
  [   11.229952] Bluetooth: hci0: Debug lock is disabled
  [   11.229952] Bluetooth: hci0: Minimum firmware build 1 week 10 2014
  [   11.230549] pstore: Using crash dump compression: deflate
  [   11.230551] pstore: Registered efi as persistent store backend
  [   11.233809] bluetooth hci0: Direct firmware load for intel/ibt-20-1-3.sfi failed with e  rror -2
  [   11.233810] Bluetooth: hci0: Failed to load Intel firmware file (-2)
```

**Expected behavior**
kernel module to be loaded

**Additional context**
New Laptop NixOs installation

**Metadata**

## nix run nixpkgs.nix-info -c nix-info -m
```sh
 - system: `"x86_64-linux"`
 - host os: `Linux 5.2.14, NixOS, 19.03.173489.af5481e9b77 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.03.173489.af5481e9b77"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by mudrii** [comment]
I tried the latest Arch LiveCD 2019.09 with slightly older kernel and both Wifi and Bluetooth works

Dmesg output

```sh
iwlwifi 0000:52:00.0: enabling device (0000 -> 0002)
iwlwifi 0000:52:00.0: Found debug destination: EXTERNAL_DRAM
iwlwifi 0000:52:00.0: Found debug configuration: 0
iwlwifi 0000:52:00.0: loaded firmware version 48.954cff6d.0 op_mode iwlmvm
iwlwifi 0000:52:00.0: Detected Intel(R) Wi-Fi 6 AX200 160MHz, REV=0x340
iwlwifi 0000:52:00.0: Applying debug destination EXTERNAL_DRAM
iwlwifi 0000:52:00.0: Allocated 0x00400000 bytes for firmware monitor.
iwlwifi 0000:52:00.0: base HW address: dc:71:96:f0:ed:ad
iwlwifi 0000:52:00.0 wlp82s0: renamed from wlan0
```

```sh
Bluetooth: Core ver 2.22
Bluetooth: HCI device and connection manager initialized
Bluetooth: HCI socket layer initialized
Bluetooth: L2CAP socket layer initialized
Bluetooth: SCO socket layer initialized
Bluetooth: hci0: Bootloader revision 0.3 build 0 week 24 2017
Bluetooth: hci0: Device revision is 1
Bluetooth: hci0: Secure boot is enabled
Bluetooth: hci0: OTP lock is enabled
Bluetooth: hci0: API lock is enabled
Bluetooth: hci0: Debug lock is disabled
Bluetooth: hci0: Minimum firmware build 1 week 10 2014
Bluetooth: hci0: Found device firmware: intel/ibt-20-1-3.sfi
Bluetooth: hci0: Waiting for firmware download to complete
Bluetooth: hci0: Firmware loaded in 1802869 usecs
Bluetooth: hci0: Waiting for device to boot
Bluetooth: hci0: Device booted in 12709 usecs
Bluetooth: hci0: Found Intel DDC parameters: intel/ibt-20-1-3.ddc
Bluetooth: hci0: Applying Intel DDC parameters completed
```
```sh
Linux archiso 5.2.11-arch1-1-ARCH #1 SMP PREEMPT Thu Aug 29 08:09:36 UTC 2019 x86_64 GNU/Linux
```

Seems NixOS Kernel does not contain latest Intel wifi drivers

**Comment by georgewhewell** [comment]
this is missing firmware, was fixed by #65091 and probably needs backporting to 19.03

**Comment by fpletz** [comment]
Fixed with https://github.com/NixOS/nixpkgs/compare/5e61b9b1f0c6...73dc83815347.
